packages/lib/src/Button.wc.svelte

```diff
@@ -1,48 +1,49 @@
 <svelte:options customElement="sesamy-button" />
 
 <script lang="ts">
-  import { twMerge } from "tailwind-merge";
-  import Base from "./Base.svelte";
+  import { twMerge } from 'tailwind-merge';
+  import Base from './Base.svelte';
+  import type { ButtonProps, Variant } from './types';
 
-  export let variant: "primary" | "secondary" = "primary";
-  export let loading = false;
-  export let disabled = false;
-  export let outline = false;
-  export let size: "sm" | "md" | "lg" = "md";
-  export let part;
-  export let onclick = () => {};
-  export let href = "";
+  let {
+    loading = false,
+    variant = 'primary',
+    disabled = false,
+    size = 'md',
+    part,
+    onclick,
+    href
+  }: ButtonProps = $props();
 
   const baseClasses =
-    "inline-flex items-center justify-center font-medium rounded-md transition-colors duration-200 ease-in-out focus:outline-none focus:ring-2 focus:ring-offset-2";
+    'inline-flex items-center justify-center font-medium rounded-md transition-colors duration-200 ease-in-out focus:outline-none focus:ring-2 focus:ring-offset-2';
 
   const sizeClasses = {
-    sm: "px-3 py-1.5 text-sm",
-    md: "px-4 py-2 text-base",
-    lg: "px-6 py-3 text-lg",
+    sm: 'px-3 py-1.5 text-sm',
+    md: 'px-4 py-2 text-base',
+    lg: 'px-6 py-3 text-lg'
   };
 
   const variantClasses = {
-    primary: {
-      solid:
-        "bg-[var(--sesamy-button-bg)] text-[var(--sesamy-button-text)] hover:bg-[var(--sesamy-button-bg-hover)] focus:ring-[var(--sesamy-button-ring)]",
-      outline:
-        "bg-transparent border border-[var(--sesamy-button-border)] text-[var(--sesamy-button-text)] hover:bg-[var(--sesamy-button-bg-hover)] hover:text-[var(--sesamy-button-text-hover)] focus:ring-[var(--sesamy-button-ring)]",
-    },
-    secondary: {
-      solid:
-        "bg-[var(--sesamy-button-secondary-bg)] text-[var(--sesamy-button-secondary-text)] hover:bg-[var(--sesamy-button-secondary-bg-hover)] focus:ring-[var(--sesamy-button-secondary-ring)]",
-      outline:
-        "bg-transparent border border-[var(--sesamy-button-secondary-border)] text-[var(--sesamy-button-secondary-text)] hover:bg-[var(--sesamy-button-secondary-bg-hover)] hover:text-[var(--sesamy-button-secondary-text-hover)] focus:ring-[var(--sesamy-button-secondary-ring)]",
-    },
+    primary:
+      'bg-[rgb(var(--s-main-color))] text-white hover:brightness-90 focus:ring-[rgb(var(--s-main-color)/0.5)]',
+    secondary:
+      'bg-transparent border border-[rgb(var(--s-main-color))] text-[rgb(var(--s-main-color))] hover:bg-[rgb(var(--s-main-color)/0.1)] focus:ring-[rgb(var(--s-main-color)/0.5)]',
```

Comment:
This is more or less the change in styling. The color needs to be wrapped in a rgb function and then we apply the opacity:
hover:bg-[rgb(var(--s-main-color)/0.1)]

Reply:
Let's sync about this exact thing!

Does `hover:bg-[rgb(var(--s-main-color)/0.1)]` really work?

Reply:
Yepp, works fine :)

Reply:
Just thinking out loud but if we want to support hex in css vars in this specific case a hack could be to put a "bg-element" inside the button.. Something like:
```
<button className="group relative">
  <span className="absolute inset-0 bg-[var(--main-color)] group-hover:opacity-50"></span>
  <span className="relative">
    <slot />
  </span>
</button>
```

Reply:
Surely something we could do.

But going back to the gradient thing I wanted to do: having colors as RGB we could do that with ease and without using a hack.

Me and Markus will soon sync on this and decide upon a solution. I have a very clear thing in mind, we just gotta choose between `rgb`, `rgba` and or `hsla`.

In worst case scenario our clients just need to adhere to one of these color standards rather than hex when using custom CSS variables to override ours.

This honestly just makes sense from a "being a modern solution"-standpoint. Hexcolors are in a way legacy and the opacity feature of them is just something that came along down the road, and wasn't a thing initially.